### PR TITLE
Center align `Loading` component based on remaining screen height

### LINKED ISSUE #8762: Reposition the page loader in main homepages to be placed in the center
**Describe the bug**

Whenever we switch tab in the main pages such as facility, patient, assets, etc... the page loader is placed in the bottom of the page

**To Reproduce**

**Expected behavior**

Adjust the position of the page loader on all tab pages (e.g., patient tab, facility tab, etc.) to appear in the center of the screen. 

**Screenshots**

Image

### PR DESCRIPTION
**Pull request by Suhail2701**
Fixes #8762 - Repositions the page loader to be centered on the all pages.

This PR updates the loader component’s CSS, ensuring that it appears in the center of the screen on all pages.

After Changes:
Loader

## Summary by CodeRabbit

- **New Features**
	- Enhanced the Loading component for improved responsiveness and adaptability based on viewport position.

- **Bug Fixes**
	- Resolved layout issues by transitioning from a grid system to a flexbox model for better alignment and centering.

- **Refactor**
	- Simplified the inner structure of the Loading component while maintaining a fixed width for the logo.

### CODE REVIEW

File: src/components/Common/Loading.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
 const img = "/images/care_logo_gray.svg";
 
 const Loading = () => {
+  const isPatientsPage = window.location.pathname === "/patients";
+  const isAssetsPage = window.location.pathname === "/assets";
+  const isUserPage = window.location.pathname === "/users";
+
+  const minHeight =
+    isPatientsPage || isAssetsPage || isUserPage ? "500px" : "800px";
```
Comment (rithviknishad) [inline src/components/Common/Loading.tsx]: hardcoding things inside a generic component based on where it's going to be used is not maintainable. 

Assume generic components to be built like a black-box. It itself shouldn't take decisions for itself based on where it'll be used, however you may pass props from the parent component for this component to behave in a particular desired way.

File: src/components/Common/Loading.tsx [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-      <div className="col-span-12 flex items-center justify-center sm:col-span-12 md:col-span-12 lg:col-span-12">
-        <div className="App">
-          <header className="App-header">
-            <img src={img} className="App-logo" alt="logo" />
-          </header>
-        </div>
+    <div
+      className="flex w-full items-center justify-center"
+      style={{ minHeight }}
+    >
```
Comment (rithviknishad) [inline src/components/Common/Loading.tsx]: let's not mix css and tailwind classes if it's solvable with tailwind classes alone

File: package-lock.json
Comment (rithviknishad) [inline package-lock.json:1]: skip checking-in changes to the lock file if no packages has been changed.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
@Suhail2701 keep readable titles for the PR and fo follow the PR template which includes the merge checklist.

**Comment by Suhail2701** [comment]
Hi @rithviknishad,

I have updated the Loading component as you mentioned. Kindly check it and let me know your feedback.

Thank you!

**Comment by bodhish** [comment]
@coderabbitai full review

File: src/components/Common/Loading.tsx [hunk 3]
```diff
@@ -1,16 +1,15 @@
 const img = "/images/care_logo_gray.svg";
 
-const Loading = () => {
+const Loading = ({ minHeightClass = "min-h-screen" }) => {
```
Comment (rithviknishad) [inline src/components/Common/Loading.tsx]:
```suggestion
const Loading = (props: { className?: string }) => {
```

File: src/components/Common/Loading.tsx [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
-    <div className="grid">
-      <div className="col-span-12 flex items-center justify-center sm:col-span-12 md:col-span-12 lg:col-span-12">
-        <div className="App">
-          <header className="App-header">
-            <img src={img} className="App-logo" alt="logo" />
-          </header>
-        </div>
+    <div
+      className={`flex w-full items-center justify-center ${minHeightClass}`}
+    >
```
Comment (rithviknishad) [inline src/components/Common/Loading.tsx]:
```suggestion
    <div
      className={cn("flex w-full items-center justify-center min-h-screen", props.className)}
    >
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

**Review by Jacobjeevan — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Pull request merged**